fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorEventProcessor.java

```diff
@@ -510,6 +516,14 @@ private void processDropPartition(DropPartitionEvent dropPartitionEvent) {
         TablePartition tablePartition =
                 new TablePartition(
                         dropPartitionEvent.getTableId(), dropPartitionEvent.getPartitionId());
+
+        // When drop a table partition, drop its snapshot store meanwhile.
```

Comment:
dito:
Only when the table is a primary key table, do we need to remove snapshot store